File: JetpackViewModelJavaVersion/app/src/main/java/com/teletian/jetpackviewmodeljavaversion/MyViewModel.java
package com.teletian.jetpackviewmodeljavaversion;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MyViewModel extends ViewModel {

    private MutableLiveData<String> news;

    public MyViewModel() {
        this.news = new MutableLiveData<>();
        fetchNews();
    }

    public MutableLiveData<String> getNews() {
        return news;
    }

    private void fetchNews() {
        new Thread(() -> {
            try {
                Thread.sleep(3000);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
            news.postValue("Completed!");
        }).start();
    }
}

File: JetpackViewModel/app/src/main/java/com/teletian/jetpackviewmodel/MyViewModel.kt
package com.teletian.jetpackviewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MyViewModel : ViewModel() {

    private val news: MutableLiveData<String> by lazy {
        MutableLiveData<String>().also {
            fetchNews()
        }
    }

    fun getNews(): LiveData<String> {
        return news
    }

    private fun fetchNews() {
        Thread {
            Thread.sleep(3000)
            news.postValue("Completed!")
        }.start()
    }
}